packages/asset-buyer/src/types.ts

```diff
@@ -25,84 +25,110 @@ export interface OrderProviderResponse {
 export interface SignedOrderWithRemainingFillableMakerAssetAmount extends SignedOrder {
     remainingFillableMakerAssetAmount?: BigNumber;
 }
+
 /**
  * gerOrdersAsync: Given an OrderProviderRequest, get an OrderProviderResponse.
  * getAvailableMakerAssetDatasAsync: Given a taker asset data string, return all availabled paired maker asset data strings.
  */
 export interface OrderProvider {
     getOrdersAsync: (orderProviderRequest: OrderProviderRequest) => Promise<OrderProviderResponse>;
     getAvailableMakerAssetDatasAsync: (takerAssetData: string) => Promise<string[]>;
+    getAvailableTakerAssetDatasAsync: (makerAssetData: string) => Promise<string[]>;
+}
+
+export interface CalldataInformation {
+    calldataHexString: string;
+    to: string;
+    value: BigNumber;
+}
+
+export interface SmartContractParams {
+    params: { [name: string]: any };
+    to: string;
+    value: BigNumber;
+}
+
+export interface SwapQuoteConsumer {
+    getCalldataOrThrow(quote: SwapQuote, opts: Partial<SwapQuoteGetOutputOpts>): CalldataInformation;
```

Comment:
Personally I'm not a fan of our `OrThrow` convention (anything can throw at the end of the day). Especially for a public-facing, top-level API, I think it's ok to omit the `OrThrow` but I'd defer to @fabioberger